server/socketManager.js

```diff
@@ -0,0 +1,334 @@
+const { Server } = require('bweb');
+const assert = require('bsert');
+const { sha256, random, ccmp } = require('bcrypto');
+const { base58 } = require('bstring');
+const { URL } = require('url');
+const Validator = require('bval');
+
+class SocketManager extends Server {
+  /**
+   * Create an http server.
+   * @constructor
+   * @param {Object} options
+   */
+  constructor(options) {
+    super(new SocketManagerOptions(options));
+
+    this.logger =
+      this.options.logger && this.options.logger.context('socket-manager');
+    this.clients = new Map();
+    this.types = ['node', 'wallet'];
+    this.init();
+  }
+
+  /*
+   * Initialize server
+   */
+  init() {
+    this.on('request', req => {
+      if (req.method === 'POST' && req.pathname === '/') return;
+
+      this.logger.debug(
```

Comment:
This class could probably be extended to be a full bPanel Server and also handle routes. Did not want to make this bigger than it already was, but shouldn't be too difficult to migrate clientRoutes to a router in this class (same as bcoin's HTTP classes)

Reply:
Would definitely prefer for this in the future to eliminate `express`